Neos.Flow/Classes/Security/Authentication/Token/UsernamePassword.php

```diff
@@ -13,30 +13,53 @@
 
 use Neos\Flow\Annotations as Flow;
 use Neos\Flow\Mvc\ActionRequest;
+use Neos\Flow\Security\Exception\InvalidAuthenticationStatusException;
 use Neos\Utility\ObjectAccess;
 
 /**
  * An authentication token used for simple username and password authentication.
  */
-class UsernamePassword extends AbstractToken
+class UsernamePassword extends AbstractToken implements UsernamePasswordInterface
 {
+    private const DEFAULT_USERNAME_POST_FIELD = '__authentication.Neos.Flow.Security.Authentication.Token.UsernamePassword.username';
+    private const DEFAULT_PASSWORD_POST_FIELD = '__authentication.Neos.Flow.Security.Authentication.Token.UsernamePassword.password';
+
     /**
      * The username/password credentials
      * @var array
      * @Flow\Transient
      */
     protected $credentials = ['username' => '', 'password' => ''];
 
+    /**
+     * The current ActionRequest
+     * @var ActionRequest
+     */
+    protected $actionRequest;
```

Comment:
That's dangerous, it will serialize the action request into the token!
(You could add a `@Flow\Transient` annotation, but IMO it would be better not to keep the request in the object state at all)